app/src/main/java/ca/yk/gov/vaxcheck/ui/scanner/BarcodeScannerFragment.kt

```diff
@@ -219,9 +216,21 @@ class BarcodeScannerFragment : Fragment(R.layout.fragment_barcode_scanner), Scan
     private suspend fun collectImmunizationStatus() {
         viewModel.status.collect { status ->
             sharedViewModel.setStatus(status)
-            findNavController().navigate(
-                R.id.action_barcodeScannerFragment_to_barcodeScanResultFragment
-            )
+            val (state, _) = status
+            when (state) {
+                VaccinationStatus.FULLY_VACCINATED,
+                VaccinationStatus.PARTIALLY_VACCINATED,
+                VaccinationStatus.NOT_VACCINATED -> {
+                    sharedViewModel.setStatus(status)
```

Comment:
you have already set the status in shared viewModel in line number 218 so you can avoid repetition.